packages/gestalt/src/SideNavigationMobile.js

```diff
@@ -1,32 +1,50 @@
 // @flow strict
-import { Fragment, type Node } from 'react';
+import { Fragment, useRef, useEffect, type Node } from 'react';
 import classnames from 'classnames';
 import Box from './Box.js';
 import Flex from './Flex.js';
 import Divider from './Divider.js';
 import Heading from './Heading.js';
-import IconButton from './IconButton.js';
 import styles from './SideNavigation.css';
 import borderStyles from './Borders.css';
-import getChildrenToArray from './getChildrenToArray.js';
+import getChildrenToArray from './SideNavigation/getChildrenToArray.js';
+import InternalDismissButton from './InternalDismissButton.js';
 import { useSideNavigation } from './contexts/SideNavigationProvider.js';
 import { type Props } from './SideNavigation.js';
 
+type SideNavigationMobileProps = {| ...Props, id: string |};
```

Comment:
nit: this would be a bit more readable if you rename the imported Props type and use "Props" here:
```js
import { type Props as SideNavigationProps } from '';

type Props = {|
  ...SideNavigationProps,
  id: string,
|};
```

Reply:
indeed

Reply:
DONE